packages/react-composites/src/composites/CallWithChatComposite/adapter/AzureCommunicationCallWithChatAdapter.ts

```diff
@@ -439,21 +446,22 @@ export class AzureCommunicationCallWithChatAdapter implements CallWithChatAdapte
   }
   /** Send a chat read receipt. */
   public async sendReadReceipt(chatMessageId: string): Promise<void> {
-    return await this.chatAdapterPromise.then((adapter) => {
+    return await this.executeWithResolvedChatAdapter((adapter) => {
       return adapter.sendReadReceipt(chatMessageId);
     });
   }
   /** Send an isTyping indicator. */
   public async sendTypingIndicator(): Promise<void> {
-    return await this.chatAdapterPromise.then((adapter) => {
+    return await this.executeWithResolvedChatAdapter((adapter) => {
       return adapter.sendTypingIndicator();
     });
   }
   /** Load previous Chat messages. */
   public async loadPreviousChatMessages(messagesToLoad: number): Promise<boolean> {
-    return await this.chatAdapterPromise.then((adapter) => {
-      return adapter.loadPreviousChatMessages(messagesToLoad);
-    });
+    if (!this.chatAdapter) {
+      throw new Error('Chat is not initialized');
```

Comment:
Out of interest, why does this not use the executeWithResolvedChatAdapter? Is there a reason this throws an error where others are logging an error?